contentcuration/contentcuration/frontend/shared/data/serverSync.js

```diff
@@ -80,6 +81,116 @@ function trimChangeForSync(change) {
   }
 }
 
+function socketHandleAllowed(allowed) {
+  if (allowed.length) {
+    const revMap = {};
+    for (let obj of allowed) {
+      revMap[obj.rev] = obj.server_rev;
+    }
+    return db[CHANGES_TABLE].where('rev')
+      .anyOf(Object.keys(revMap).map(Number))
+      .modify(c => {
+        c.server_rev = revMap[c.rev];
+        c.synced = true;
+      });
+  }
+  return Promise.resolve();
+}
+function socketHandleDisallowed(disallowed) {
+  if (disallowed.length) {
+    // Collect all disallowed
+    const disallowedRevs = disallowed.map(d => Number(d.rev));
+    // Set the return error data onto the changes - this will update the change
+    // both with any errors and the results of any merging that happened prior
+    // to the sync operation being called
+    return db[CHANGES_TABLE].where('rev')
+      .anyOf(disallowedRevs)
+      .modify({ disallowed: true, synced: true });
+  }
+  return Promise.resolve();
+}
+
+function socketHandleReturnedChanges(returnedChange) {
+  // The changes property is an array of any changes from the server to apply in the
+  // client.
+  let returnedChanges = [];
+  returnedChanges.push(returnedChange);
+  if (returnedChanges.length) {
+    return applyChanges(returnedChanges);
+  }
+  return Promise.resolve();
+}
```

Comment:
I don't think this function is needed. I think the call to this can be replaced with `applyChanges([data.change])`